src/pages/en/guides/content-collections.mdx

```diff
@@ -187,290 +173,250 @@ defineCollection({
+```json title="tsconfig.json" ins={3}
+{
+  "compilerOptions": {
+    "strictNullChecks": true
+  }
+}
+```
 
-For example, to use a frontmatter `permalink` property as the slug for your blog pages instead of the file path, you can use the following `slug()` function. For extra safety, conditionally return the entry's `defaultSlug` as a fallback.
+Note that if you are currently extending Astro's `strict` or `strictest` recommended settings, no changes is needed.
 
-```ts {5-9}
-// src/content/config.ts
-import { defineCollection, z } from 'astro:content';
+```json title="tsconfig.json" "strict" "strictest"
+{
+  // No change needed!
+  "extends": "astro/tsconfigs/strict"
+}
+```
 
-const blog = defineCollection({
-  slug: ({ id, defaultSlug, data, body }) => {
-    // Use `permalink` from the entry’s frontmatter as the slug, if it exists.
-    // Otherwise, fall back to the default slug.
-    return data.permalink || defaultSlug;
-  },
-  schema: z.object({
-    permalink: z.string().optional(),
-  }),
-});
+### The ".astro" Directory
 
-export const collections = { blog };
-```
+Astro generates type information and metadata from your content collections in a new `.astro` directory. No action is needed on your part to maintain or update this directory. It is safe to ignore it in your project.
 
-## Querying content collections
+:::tip
+If you're using Git for version control, we recommend ignoring this generated directory by adding `.astro` to your `.gitignore`.
+:::
 
-Astro provides two functions to query collections:
+The `.astro` directory will be updated for you automatically anytime you run the [`astro dev`](/en/reference/cli-reference/#astro-dev), [`astro build`](/en/reference/cli-reference/#astro-build) commands. You can run [`astro sync`](/en/reference/cli-reference/#astro-sync) yourself to update `.astro` manually.
 
-### `getCollection()`
 
-`getCollection()` returns multiple entries in a collection. It requires the name of a `collection` as a parameter.  By default, it returns all items in the collection.
+## The "src/content" Directory
 
-It can also take a second, optional parameter: a filter function based on schema properties. This allows you to query for only some items in a collection based on `id`, `slug`, or frontmatter values via the `data` object.
+Astro treats the `src/content/` directory as special. This is where **collections** (folders) of Markdown/MDX **entries** (files) can be stored, with a single configuration file to define each collection's **schema** (frontmatter data types and shape). Files other than your `.md`/`.mdx` content are not permitted inside `src/content/`.
 
-```astro
----
-import { getCollection } from 'astro:content';
 
-// Get all `src/content/blog/` entries
-const allBlogPosts = await getCollection('blog');
+## Querying Collections
```

Comment:
From this section, to Generating Routes, I'm not sure the heading levels/right sidebar organization is quite right.

Content-wise, Rendering Content doesn't seem like it should be a `##` as it seems more a peer with "Using Content in Astro Templates". But, I'm not sure I'd throw *everything* under the title of "Querying Collections"? And, they alternate between "script stuff" and "templating stuff".

These related items seem to be:

Script - Making the queries/filtering
Template- Templating the metadata (but not using `render()`)
Script - Passing an entire Content as props
Template - Rendering the content itself with `render()`
Script/Template of modified metadata

I think these could all be `###` peers under one `##`? (Which we'd need a name for.)

Reply:
I'm not sure I entirely follow this, but I agree this feels a little incorrectly placed. Just reorganized it to be a party of "querying", lmk what you think.